server/src/database/setup.ts

```diff
@@ -0,0 +1,70 @@
+import { readFile } from "fs";
+import parentLogger from "../../utils/logger";
+import db from "./db";
+import { CourseCode, PreReq, CoReq, CourseSection } from "./schema";
+
+const log = parentLogger.child({ module: "router" });
+const src = "./utils/output.json";
+
+export const setupDb = async () => {
+    await dropDb();
+    await createDb();
+    await populateDb();
+};
+
+const dropDb = () => db.query(`DROP TABLE IF EXISTS CourseCode, PreReq, CoReq, CourseSection;`);
+
+const createDb = async () => {
+    await db.query(CourseCode);
+    await db.query(PreReq);
+    await db.query(CoReq);
+    await db.query(CourseSection);
+};
+
+const populateDb = async () => {
+    readFile(src, { encoding: "utf8" }, async (err: any, data: any) => {
+        if (err) return;
+        const courses = JSON.parse(data);
+        log.info(`${courses.length} courses`);
+        const coReqsToStore: any[] = [];
+        const preReqsToStore: any[] = [];
+        const sectionsToStore: any[] = [];
+        courses.forEach((course: any) => {
+            const { courseTitle, courseCode, preReqs = [], coReqs = [], sections = [] } = course;
+            const [courseDept, courseNumber] = courseCode.split(" ");
+            coReqs.forEach((coReq: any) => {
+                const [coReqCourseDept, coReqCourseNumber] = coReq.split(" ");
+                coReqsToStore.push([courseDept, courseNumber, coReqCourseDept, coReqCourseNumber]);
+            });
+            preReqs.forEach(async (preReq: any) => {
+                const [preReqCourseDept, preReqCourseNumber] = preReq.split(" ");
+                preReqsToStore.push([courseDept, courseNumber, preReqCourseDept, preReqCourseNumber]);
+            });
+            sections.forEach(async (section: any) => {
+                const { sectionTitle, status, activity, prof, timeInfo } = section;
+                const { term, day, start, end } = timeInfo;
+                // TODO: handle cases where start or end is an empty string
+                sectionsToStore.push([courseTitle, courseDept, courseNumber, sectionTitle, status, activity, prof, term, day]);
+            });
+        });
+        try {
+            log.info(`Found ${coReqsToStore.length} co-requisites.`);
+            log.info(`Found ${preReqsToStore.length} pre-requisites.`);
+            log.info(`Found ${sectionsToStore.length} course sections.`);
+            preReqsToStore.forEach(async ([courseDept, courseNumber, preReqCourseDept, preReqCourseNumber]) => {
```

Comment:
I think we need to use `.map()` here and call `await Promise.all()` on the resulting array b/c  I think each of these promises are not be awaited on before the subsequent queries and logs on L59-65.

Reply:
ForEach can cause some weird bugs, but I assumed this code worked because of the screenshots and I think @yehee tested with a bunch of queries.

Reply:
My bad! They initially had a foreign key constraint which has been taken out for now, so should fine to do what Jackson has suggested. In fact we probably don't need to keep `CourseCode` table around, wondering how others think?

Reply:
Hmm, if we remove it, wouldn't we lose the data for courses that don't have any sections, pre-reqs, co-reqs? 
Whether those courses exist or not is a different concern though haha

Reply:
just chiming in to say that there arent any courses without sections, but there are def courses without prereqs/coreqs of any kind

Reply:
@jacksonyuanjx Im not sure how we would lose data? @yehee I'd say it's up to you on how you wanna do the schema, if it's easier to get rid of course codes then we can do it.

Reply:
@jacksonyuanjx Ohh I see, not sure when that would be the case but should also take those into consideration.
@resurreccionl Good to know! Those that are missing pre-reqs/co-reqs are getting inserted as well so should be fine I think.

Reply:
@johnagl didn't see your comment when I wrote my comment above! I'd say we leave it out for now for simplicity, I think we can always add those back if we ever need those for any reasons moving forward.